public/app/core/components/AppChrome/TopBar/SingleTopBar.tsx

```diff
@@ -98,11 +104,40 @@ export const SingleTopBar = memo(function SingleTopBar({
           <TopSearchBarCommandPaletteTrigger />
           {unifiedHistoryEnabled && !isSmallScreen && <HistoryContainer />}
           {!isSmallScreen && <QuickAdd />}
-          {enrichedHelpNode && (
-            <Dropdown overlay={() => <TopNavBarMenu node={enrichedHelpNode} />} placement="bottom-end">
-              <ToolbarButton iconOnly icon="question-circle" aria-label={t('navigation.help.aria-label', 'Help')} />
-            </Dropdown>
-          )}
+          {enrichedHelpNode &&
+            (enrichedHelpNode.hideFromTabs && availableComponents.has('grafana-grafanadocsplugin-app') ? (
+              (() => {
+                const componentId = getComponentIdFromComponentMeta('grafana-grafanadocsplugin-app', {
+                  title: 'Grafana Pathfinder',
+                } as any);
+                const isOpen = dockedComponentId === componentId;
+                return (
+                  <ToolbarButton
+                    iconOnly
+                    icon="question-circle"
+                    aria-label={t('navigation.help.aria-label', 'Help')}
+                    className={isOpen ? styles.helpButtonActive : undefined}
+                    onClick={() => {
+                      if (isOpen) {
+                        setDockedComponentId(undefined);
+                      } else {
+                        const appEvents = getAppEvents();
+                        appEvents.publish(
+                          new OpenExtensionSidebarEvent({
+                            pluginId: 'grafana-grafanadocsplugin-app',
+                            componentTitle: 'Grafana Pathfinder',
+                          })
+                        );
+                      }
+                    }}
+                  />
+                );
+              })()
+            ) : (
+              <Dropdown overlay={() => <TopNavBarMenu node={enrichedHelpNode} />} placement="bottom-end">
+                <ToolbarButton iconOnly icon="question-circle" aria-label={t('navigation.help.aria-label', 'Help')} />
+              </Dropdown>
+            ))}
           <NavToolbarSeparator />
```

Comment:
felt this big ? operator here and the code inside the first path make this markup a bit hard to read. 

Pushed a propsed refactoring here:
https://github.com/grafana/grafana/compare/jdb/2025-09-docs-plugin-menu...docs-help-plugin?expand=1

Also get's rid of the `as any` typecast